README.md



* User:  " iniciar secion con estas credenciale"
```bash
admin@mkthub.tech
```
* Password: 
```bash
admin
```
   






docs/CHANGELOG.md
# Changelog

<<<<<<< HEAD
### 23/12/2022

- [Update in view of all ticket for all users](https://github.com/FilipeCamillo/Press-Ticket/commit/708977b71d0a6d430e522910cd4e836519e1a70a)
=======
### 24/12/2022

- [Atualização do ListTicketService](https://github.com/rtenorioh/Press-Ticket/commit/6bbd2f071a4a8af9c2c6e132f1379538e7ecf3f7);
- [Inclusão de validação no código para exibir o número conectado](https://github.com/rtenorioh/Press-Ticket/commit/911865958ffe9b0ce3eed7fe2085779706ebaefd); e
- [Correção da variável da porta do Frontend](https://github.com/rtenorioh/Press-Ticket/commit/0bcc18ca1c1364ffde34660eea7dc8c014a56a86).

### 29/11/2022

- [Inclusão do Número da Conexão](https://github.com/rtenorioh/Press-Ticket/commit/5827a45362f9824a1d1ee5dceb030dac7509f305).

### 27/11/2022

- [Atualização da lib whatsapp-web.js](https://github.com/rtenorioh/Press-Ticket/commit/75fd718dea270df43b59249ee291470d4220648a).
>>>>>>> 0e65719eec11637c851495e6065a70e38bc927f3

### 15/11/2022

- [Atualização das informações](https://github.com/rtenorioh/Press-Ticket/commit/e62551c256bd0a66f1bc2a773415a80623a1f7e5).

### 14/11/2022

- [Atualiação das libs: wwebjs e pino-pretty](https://github.com/rtenorioh/Press-Ticket/commit/0351089f1911253e515aaea712ebe236c5c26a11).

### 11/11/2022

- [Ajuste para verificação do config.json](https://github.com/rtenorioh/Press-Ticket/commit/95ba26e1fbdccc68bf1d36815b1c88becacf5d7e); e
- [Correção na tradução](https://github.com/rtenorioh/Press-Ticket/commit/e0f9a6324a9006b8484463cf82838467f347ce81).

### 07/11/2022

- [Alteração na estrutura de cores](https://github.com/rtenorioh/Press-Ticket/commit/4e4a8930b340cf02ef522ddb817760b7bc7f7fe4);
- [Inclusão do import](https://github.com/rtenorioh/Press-Ticket/commit/d05086e4f8af2560bb1bbe38044cf4348bf41c3b); e
- [Atualização da versão](https://github.com/rtenorioh/Press-Ticket/commit/98f7fd33e0c1d1e4a228dda891fa84777c617bee).

### 06/11/2022

- [Inclusão do Filtro de Tags](https://github.com/rtenorioh/Press-Ticket/commit/5e6a8747db614879ba68aea00c831fd653998899).

### 23/10/2022

- [Configuração da exibição dos horários dos setores](https://github.com/rtenorioh/Press-Ticket/commit/5bc69b6832aada42f39aadfec0b8c171f7b3445a); 
- [Função copiar dados do contato](https://github.com/rtenorioh/Press-Ticket/commit/a4f72cfff0df33696bd9a1bcf795cd77fa802d93); e
- [Atualização da versão](https://github.com/rtenorioh/Press-Ticket/commit/7ce64f1333c520d672266633a46eb1f6ae24e5bf).

### 20/10/2022

- [Correção da marca de leitura da mensagem](https://github.com/rtenorioh/Press-Ticket/commit/6e68ec866104a11f90f52ae9d635899f186a1bab).

### 19/10/2022

- [Horário de funcionamento do Atendente](https://github.com/rtenorioh/Press-Ticket/commit/404b61ae332d9b66b3d18bd9afa112378d9f8f3a); e
- [Atualização da versão](https://github.com/rtenorioh/Press-Ticket/commit/aab58ad0b6beea430a381d94af47cfb3705d6809).

### 18/10/2022

- [Correção](https://github.com/rtenorioh/Press-Ticket/commit/ba92c2fb4263f28bc329765128d4169f950a8e7f).

### 13/10/2022

- [Remoção do console.log](https://github.com/rtenorioh/Press-Ticket/commit/75d64af0b63b05ff753aa30226b8847ec8a3948a).

### 12/10/2022

- [Correção nos comandos](https://github.com/rtenorioh/Press-Ticket/commit/a580d599e92c6de19dcdd498378dc3187a973beb);
- [Correção na tradução da transferência de setor](https://github.com/rtenorioh/Press-Ticket/commit/1291369d633a7ca1f0d0a5bb914371f1e2039655);
- [Correção na tradução ao deletar ticket](https://github.com/rtenorioh/Press-Ticket/commit/7e8436fa11740d1f6271b3d321165ca763eb69c6); e
- [Botão aceitar apenas para Admin ao espiar](https://github.com/rtenorioh/Press-Ticket/commit/4989417415efb0ba91c4e005a51d70b32a1f1f79).


### 09/10/2022

- [Correção](https://github.com/rtenorioh/Press-Ticket/commit/4db7f7aab35bff04d9b1b2475fdc3b0637ff7bfc);
- [Ajustes no layout](https://github.com/rtenorioh/Press-Ticket/commit/e2d239b670080a90887afd8e00eb702fb578a9a3);
- [Aumento no carregamento dos contatos para 5000](https://github.com/rtenorioh/Press-Ticket/commit/f97cc726104f5dfe58f68278ae4731d805090ec2); e
- [Atualização da versão](https://github.com/rtenorioh/Press-Ticket/commit/2d235da25aa7f16dd94460e435e34a422622bee2).

### 08/10/2022

- [Create ModalImage.js](https://github.com/rtenorioh/Press-Ticket/commit/724e55c42448fc999f4b47b0fcdfd4655417f37d); e
- [Visualização maior do contato](https://github.com/rtenorioh/Press-Ticket/commit/024f916fead89311683d9f567efdfec9a2b37914).

### 07/10/2022

- [Arrastar e soltar medias e enviar com ENTER](https://github.com/rtenorioh/Press-Ticket/commit/32d434801c85b1b1ca342a0362daf1bc8e57e5c1).

### 28/09/2022

- [Inclusão do contador de uso das Tags](https://github.com/rtenorioh/Press-Ticket/commit/1178a204cf75678cf689bd7ba7ea49dcf78c5bee).

### 27/09/2022

- [Tags no Contato](https://github.com/rtenorioh/Press-Ticket/commit/7c3357a1cce91f1c38273a83347f733a6f832fa1).

### 25/09/2022

- [Correção da renderização ao responder uma mensagem](https://github.com/rtenorioh/Press-Ticket/commit/e2dbc79f43c1fc621b4d924a9c4c030360e891f0).

### 22/09/2022

- [Retirada do envio da despedida para grupos](https://github.com/rtenorioh/Press-Ticket/commit/7f5a5fa00d9f349a8bbfebdac9ab89778975279a).
### 21/09/2022

- [Correção na tradução do modal de tranferir usuário](https://github.com/rtenorioh/Press-Ticket/commit/aefec1cad68f7e71025be67639bbf3445d67c8a2); e
- [Correção na tradução do modal de deletar ticket](https://github.com/rtenorioh/Press-Ticket/commit/2d549e7abaab0fae2f1b417a2f3bb2f2907fefff).

### 18/09/2022

- [Atualização do typescript](https://github.com/rtenorioh/Press-Ticket/commit/94ee68250f44100050301bd79b39fc9d45ab985a).

### 15/09/2022

- [Adcionado a função de refresh](https://github.com/rtenorioh/Press-Ticket/commit/199ba219c738a5534c323411361a95cb52dbd72d);
- [Correção ao responder grupos](https://github.com/rtenorioh/Press-Ticket/commit/9721da147cd7916c1b2577f20bdd549a1bd91edc);

frontend/src/pages/Login/index.js
import React, { useState, useContext } from "react";
import { Link as RouterLink } from "react-router-dom";

import {
  Button,
  CssBaseline,
  TextField,
  Grid,
  Box,
  Typography,
  Container,
  InputAdornment,
  IconButton,
  Link
} from '@material-ui/core';

import { Visibility, VisibilityOff } from '@material-ui/icons';
import { makeStyles } from "@material-ui/core/styles";
import { i18n } from "../../translate/i18n";

import { AuthContext } from "../../context/Auth/AuthContext";
import { system } from "../../config.json";
import logo from '../../assets/logo.png';

const Copyright = () => {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      © {new Date().getFullYear()}
      {" - "}
      <Link color="inherit" href={system.url}>
        {system.name}
      </Link>
      {"."}
    </Typography>
  );
};

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

const Login = () => {
  const classes = useStyles();

  const [user, setUser] = useState({ email: "", password: "" });
  const [showPassword, setShowPassword] = useState(false);

  const { handleLogin } = useContext(AuthContext);

  const handleChangeInput = (e) => {
    setUser({ ...user, [e.target.name]: e.target.value });
  };

  const handlSubmit = (e) => {
    e.preventDefault();
    handleLogin(user);
  };

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <img alt="logo" src={logo}></img>
        <Typography component="h1" variant="h5">
          {i18n.t("iniciar secion ")}
        </Typography>
        <form className={classes.form} noValidate onSubmit={handlSubmit}>
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="email"
            label={i18n.t("email")}
            name="email"
            value={user.email}
            onChange={handleChangeInput}
            autoComplete="email"
            autoFocus
          />
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="password"
            label={i18n.t("password")}
            id="password"
            value={user.password}
            onChange={handleChangeInput}
            autoComplete="current-password"
            type={showPassword ? 'text' : 'password'}
            InputProps={{
              endAdornment: (
                <InputAdornment position="end">
                  <IconButton
                    aria-label="toggle password visibility"
                    onClick={() => setShowPassword((e) => !e)}
                  >
                    {showPassword ? <VisibilityOff color="secondary" /> : <Visibility color="secondary" />}
                  </IconButton>
                </InputAdornment>
              )
            }}
          />
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            {i18n.t("login")}
          </Button>
          <Grid container>
            <Grid item>
              <Link
                href="#"
                variant="body2"
                component={RouterLink}
                to="/signup"
              >
                {i18n.t("register")}
              </Link>
            </Grid>
          </Grid>
        </form>
      </div>
      <Box mt={8}><Copyright /></Box>
    </Container>
  );
};

export default Login;


frontend/src/pages/Dashboard/Chart.js
import React, { useState, useEffect, useRef } from "react";
import { useTheme } from "@material-ui/core/styles";
import {
	BarChart,
	CartesianGrid,
	Bar,
	XAxis,
	YAxis,
	Label,
	ResponsiveContainer,
} from "recharts";
import { startOfHour, parseISO, format } from "date-fns";

import { i18n } from "../../translate/i18n";

import Title from "./Title";
import useTickets from "../../hooks/useTickets";

const Chart = () => {
	const theme = useTheme();

	const date = useRef(new Date().toISOString());
	const { tickets } = useTickets({ date: date.current });

	const [chartData, setChartData] = useState([
		{ time: "00:00", amount: 0 },
		{ time: "01:00", amount: 0 },
		{ time: "02:00", amount: 0 },
		{ time: "03:00", amount: 0 },
		{ time: "04:00", amount: 0 },
		{ time: "05:00", amount: 0 },
		{ time: "06:00", amount: 0 },
		{ time: "07:00", amount: 0 },
		{ time: "08:00", amount: 0 },
		{ time: "09:00", amount: 0 },
		{ time: "10:00", amount: 0 },
		{ time: "11:00", amount: 0 },
		{ time: "12:00", amount: 0 },
		{ time: "13:00", amount: 0 },
		{ time: "14:00", amount: 0 },
		{ time: "15:00", amount: 0 },
		{ time: "16:00", amount: 0 },
		{ time: "17:00", amount: 0 },
		{ time: "18:00", amount: 0 },
		{ time: "19:00", amount: 0 },
		{ time: "20:00", amount: 0 },
		{ time: "21:00", amount: 0 },
		{ time: "22:00", amount: 0 },
		{ time: "23:00", amount: 0 },
	]);

	useEffect(() => {
		setChartData(prevState => {
			let aux = [...prevState];

			aux.forEach(a => {
				tickets.forEach(ticket => {
					format(startOfHour(parseISO(ticket.createdAt)), "HH:mm") === a.time &&
						a.amount++;
				});
			});

			return aux;
		});
	}, [tickets]);

	return (
		<React.Fragment>
			<Title>{`${i18n.t("dashboard.charts.perDay.title")}${
				tickets.length
			}`}</Title>
			<ResponsiveContainer>
				<BarChart
					data={chartData}
					barSize={40}
					width={730}
					height={250}
					margin={{
						top: 16,
						right: 16,
						bottom: 0,
						left: 24,
					}}
				>
					<CartesianGrid strokeDasharray="3 3" />
					<XAxis dataKey="time" stroke={theme.palette.text.secondary} />
					<YAxis
						type="number"
						allowDecimals={false}
						stroke={theme.palette.text.secondary}
					>
						<Label
							angle={270}
							position="left"
							style={{ textAnchor: "middle", fill: theme.palette.text.primary }}
						>
							Chamados
						</Label>
					</YAxis>
					<Bar dataKey="amount" fill={theme.palette.primary.main} />
				</BarChart>
			</ResponsiveContainer>
		</React.Fragment>
	);
};

export default Chart;
